src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3591,6 +3607,15 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
                   case _ => tp
                 }
 
+                // Inline RHS of ValDef for right-associative by-value operator desugaring
+                val args2 = (args1, mt.params) match {
+                  case ((ident: Ident) :: Nil, param :: Nil) if param.isByNameParam && rightAssocValDefs.contains(ident.symbol) =>
+                    inlinedRightAssocValDefs += ident.symbol
+                    val rhs = rightAssocValDefs(ident.symbol)
```

Comment:
> Are you sure?

To me it feels more like keeping state local, but it's fine either way in the end.



> It stores the RHS which is needed in doTypedApply

Right, of course, I missed that.